### PR TITLE
Add Auto fixes to ci

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

When we modify code in the browser or get a PullRequest from someone else and there is an error in the pre-commit CI, we have to launch a local terminal to do the pre-commit. This is very annoying; since the pre-commit change is being made in CI, let's commit there.

 

### Details

- None

### CODE REVIEW

File: .github/workflows/style_docstr.yml [hunk 1]
```diff
@@ -25,6 +25,11 @@ jobs:
       - name: Run pre-commit
         run: pre-commit run --all-files
```
Comment (tkoyama010) [inline .github/workflows/style_docstr.yml:26]:
```suggestion
        continue-on-error: true
        run: pre-commit run --all-files
```

File: pyvista/core/cell.py [hunk 2]
```diff
@@ -19,8 +19,6 @@ class Cell(_vtk.vtkGenericCell, DataObject):
     be useful when walking through a cell's individual faces or investigating
     cell properties.
 
-    Parameters
-    ----------
```
Comment (tkoyama010) [inline pyvista/core/cell.py:23]: Parameters
    ----------

File: pyvista/core/cell.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class Cell(_vtk.vtkGenericCell, DataObject):
     be useful when walking through a cell's individual faces or investigating
     cell properties.
 
-    Parameters
+    Prameters
```
Comment (tkoyama010) [inline pyvista/core/cell.py:22]:
```suggestion
    Parameters
```

**Comment by adeak** [comment]
I'm a bit conflicted about this. It seeems like really dangerous territory to have CI push commits. On the other hand I acknowledge that this would make web-based development a lot smoother.

Two technical points:
1. Would it be possible to have this functionality as an opt-in feature via a bot? I imagine posting a comment with `@botname autoformat` or something. It would be slightly more work than autofix by CI, but it would be fully transparent and controllable.
2. Will this work on fork branches?

**Comment by tkoyama010** [comment]
Good idea. I will try it that way.

**Comment by MatthewFlamm** [comment]
> 1. Would it be possible to have this functionality as an opt-in feature via a bot?

I agree with this.  We need to balance competing objectives here.  The use case I'm thinking of is for new contributors who are not git-savvy.  I think automated changes might scare them off as they now need to pull from remote and potentially have conflicts.  It is like dumping them in the deep end right away.  If it is opt-in via bot command, we can wait until the PR is ready to merge and then run.

**Comment by MatthewFlamm** [comment]
> we have to launch a local terminal to do the pre-commit

I had used [gitpod](https://docs.pyvista.org/extras/developer_notes.html#using-gitpod-workspace)  for these kind of uses in the past, if the bottleneck is local terminal.  It does take some time to preload everything unfortunately.

**Comment by MatthewFlamm** [comment]
> Will this work on fork branches?

It _should_ work in either method on any fork branch when the "allow for edits by maintainers" (not sure on exact wording) option is enabled.  On PRs without this, we'd have to do the old fashioned way.

**Comment by akaszynski** [comment]
Full agreement that any changes should be opt-in. I've sometimes made mistakes myself (forgot to install pre-commit in a new venv).

**Comment by banesullivan** [comment]
>  if @tkoyama010's workflow often doesn't involve a terminal of some sort then this is worth pursuing.

+1 -- If it will make anyone's workflow for contributing easier, happy to have it added

**Comment by tkoyama010** [comment]
I guess using CodeSpaces #4171 resolves this problem. Closing.

**Comment by tkoyama010** [comment]
I already solved PullRequest, but I had an idea and I'd like to comment on it. If you want to optionally perform a GitHubAction, you can fulfill this need by triggering it with a label. In this case, we only need to execute the GitHubAction when the label "auto format" is added.

**Comment by adeak** [comment]
@tkoyama010 how would that work with multiple requests to auto-format during the lifetime of a PR? If you remove the label and apply it again, will it run auto-format again? Or would auto-format run each time a commit is pushed as long as the label is there?

**Comment by tkoyama010** [comment]
@adeak
Thank you for your comments, I will stop considering this matter as [it seems complicated to deal with TOKEN](https://github.com/pre-commit/action#using-this-action-with-custom-invocations).
As a bonus to my research, I found a [pre-commit official GitHubAction](https://github.com/pre-commit/action) and created #4263.

**Pull request closed without merging**